[IR RX] Disable Extended decoding of AC protocols by default

jimmys01 opened the pull request:
Saves 20K of flash memmory. They provide no functionality other than giving log stirngs of that short:  
`Mesg Desc.: Power: On, Fan: 5 (AUTO), Mode: 3 (HEAT), Temp: 22C, Zone Follow: Off, Sensor Temp: Ignored`

jimmys01 committed:
[IR RX] Disable Extended decoding of AC protocols by default

Saves 20K of flash memmory. They provide no functionality other than giving log stirngs of that short:  
"Mesg Desc.: Power: On, Fan: 5 (AUTO), Mode: 3 (HEAT), Temp: 22C, Zone Follow: Off, Sensor Temp: Ignored"

jimmys01 committed:
[IR RX] Disable Extended decoding of AC protocols by default

Saves 20K of flash memmory. They provide no functionality other than giving log stirngs of that short:  
"Mesg Desc.: Power: On, Fan: 5 (AUTO), Mode: 3 (HEAT), Temp: 22C, Zone Follow: Off, Sensor Temp: Ignored"

jimmys01 committed:
Fix endIf statement

jimmys01 committed:
Fix endIf statement

TD-er committed:
[ESP32] Disable Gratuitous ARP for ESP32 (#2374)

See #2374

TD-er committed:
[ADC] Use simple filter for ADC plugin and show value on setup page

When using 'Oversampling' the Analog input plugin is simply summing all samples and upon calling the read function it will divide this sum by the number of samples.
This added filtering is keeping track of the lowest and highest value measured in this loop and excludes these extremes.
This makes the reading a lot more stable, since it is often only a single extreme value which causes noise in the reported values.

Also added a note to the setup page near the calibration points to make it easier to add calibration values. It shows the current raw value + its calibrated equivalent (if enabled).

jimmys01 commented:
I don't know why it is not building,

TD-er commented:
There are 2 issues here.
One is the same reason why my PR is not building and that's a change made today in the core 2.6.0 code (we're on the master branch for that one).
I already commented on that merge.
The second one is the number of #ifdef added does not match the number of #endif you added.
So without looking at the Travis output, I would say that's suspicious to say the least.

jimmys01 commented:
There are #if that perhaps you haven't counted

TD-er commented:
I now see you did a rename of a comment, that's why the diff showed it as an addition.

Just looked at the Travis output and indeed it is related to the core 2.6.0 update.
See my comment here: https://github.com/esp8266/Arduino/commit/b1da9eda467cc935307d553692fdde2e670db258#r32622483

TD-er committed:
[Travis] Change needed due to change in core 2.6.0 File::write

See   https://github.com/esp8266/Arduino/commit/b1da9eda467cc935307d553692fdde2e670db258#r32622483

TD-er committed:
[Travis] Changes needed due to update core 2.6.0 for File::write

TD-er committed:
[Build] Add build stats to track 

See also [PlatformIO docs](http://docs.platformio.org/en/latest/faq.html#program-memory-usage) and https://github.com/esp8266/Arduino/issues/3740#issuecomment-338416331

TD-er committed:
[Memory] Add several F() macro markers and disable some debug logs

All in order to reduce memory usage.

TD-er committed:
[Modbus] Split Modbus RTU from Senseair plugin

To allow other plugins to use Modbus RTU also.

TD-er committed:
[Travis] Build error including F-macro in private class webserver

TD-er committed:
[ADC] Show calibration range + step size

This min/max/stepsize info makes it easier to set the calibration.

TD-er committed:
[Travis] F-macro related build issues

Some core version builds were not using the same disable debug flags.

TD-er committed:
[MPU6050] Reverted to old version before adding Plugin_task_data array

See issue: #2381
Commits:
- https://github.com/letscontrolit/ESPEasy/commit/af20984079d3e7aa59e08fd9b232f6d17ba3b523#diff-ec860ac195fffa61ec11dd419fefa5b9
- https://github.com/letscontrolit/ESPEasy/commit/6400c495e24f39ebac88eb634f29cfb73137fa2b#diff-ec860ac195fffa61ec11dd419fefa5b9

TD-er committed:
[Cleanup] Remove P030 BMP280 since it has become obsolete by BMx280

TD-er committed:
[Webserver] Add documentation link to Advanced settings page

And output the new settings in the info page & JSON output
Not yet included in the JSON used for the new UI.

TD-er committed:
[ADC] Increase readability of ADC calibration settings

Output example:
Current:	405 ≙ 13.034
Minimum:	0 ≙ 1.664
Maximum:	1023 ≙ 30.346
Step size:	1 ≙ 0.028

jimmys01 committed:
Travis rebuild

jimmys01 committed:
Travis rebuild

TD-er commented:
A rebuild will not work yet, since I've not merged my own code.
I will do it now and then trigger a rebuild.

TD-er committed:
Merge pull request #2377 from TD-er/feature/ADC_filter

[ADC] Use simple filter for ADC plugin and show value on setup page

TD-er committed:
Merge pull request #2376 from TD-er/bugfix/2374_gratARP_esp32

[ESP32] Disable Gratuitous ARP for ESP32 (#2374)

TD-er committed:
Merge remote-tracking branch 'jimmys01/patch-2' into patch-2

TD-er commented:
I did a rebase on this PR, but now it looks like it contains a lot more commits.
I will let Travis run it and see if it will merge it.
If not, then I will create a new branch on top of what is now Mega and then cherrypick your commits, make a new PR and merge that one (if Travis agrees)
